### PR TITLE
Bucket failing Xml trimming test apps

### LINKED ISSUE #101228: Go over trimming tests we disabled for AOT
We need to go over DisabledNativeAotTestAppProjects added in #101229 and either exclude tests permanently (e.g. dispatch proxy), or figure out why the test is failing and either fix the test, or file product issues.

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Resolves #101228.

### CODE REVIEW

File: src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj [hunk 1]
```diff
@@ -2,12 +2,22 @@
...
 
   <ItemGroup>
-    <TestConsoleAppSourceFiles Include="XmlSchema.Write.cs" />
-    <TestConsoleAppSourceFiles Include="XmlSerializer.Deserialize.cs" />
+    <TestConsoleAppSourceFiles Include="XmlSchema.Write.cs">
+      <!-- What the test is testing is RUC and there was a half-hearted attempt to make it safe, but it isn't: https://github.com/dotnet/runtime/blob/5f067ce8b50087e032d13a1b97ae5ec39fc54739/src/libraries/System.Private.Xml/src/System/Xml/Schema/XmlSchema.cs#L134-L137 -->
+      <NativeAotIncompatible>true</NativeAotIncompatible>
+    </TestConsoleAppSourceFiles>
+    <TestConsoleAppSourceFiles Include="XmlSerializer.Deserialize.cs">
+      <NativeAotIncompatible>true</NativeAotIncompatible>
```
Comment (eerhardt) [inline src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj:10]: Should we list a reason for `XmlSerializer.Deserialize.cs` and `XmlSerializer.Serialize.cs`?

File: src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj [hunk 2]
```diff
@@ -2,12 +2,22 @@
...
+    </TestConsoleAppSourceFiles>
     <TestConsoleAppSourceFiles Include="XmlSerializer.Deserialize.SealerOpt.cs"
                                ExtraTrimmerArgs="--enable-opt sealer" />
```
Comment (eerhardt) [inline src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj:13]: The ExtraTrimmerArgs is kind of odd in the context of native AOT. Should this test be disabled too?

Comment (MichalStrehovsky) [reply src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj:13]: Sealing is an optimization that native AOT does automatically so it doesn't seem too irrelevant. Native AOT's version should not really be observable.

File: src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj (hunk 1 above)
Comment (MichalStrehovsky) [reply src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj:10]: I struggled with what to put there and still struggle. I don't understand the reason we have this test in the first place. If we add a new optimization to IL Link that will make this test break same as it's broken with native AOT, would we consider that "illegal" optimization? What is the purpose of trimming tests for something that is marked as trim unsafe?

(My idea always was that we write trimming test if something is annotated as trim safe, but there's a warning suppression in place.)

Comment (eerhardt) [reply src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj:10]: >  I don't understand the reason we have this test in the first place.

It is for Xamarin / Blazor WASM style trimming. See https://github.com/dotnet/runtime/issues/41389. Even though the API is marked as "unsafe" - we wanted to ensure it works if the developer does the work to preserve/mark their own types. (Sort of like how DiagnosticSource and EventSource work.) It is ensuring that we have our internals annotated correctly.

> I struggled with what to put there and still struggle.

What's the current problem we run into on native AOT? If we annotated in the test more, would that help?

Comment (MichalStrehovsky) [reply src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj]: > What's the current problem we run into on native AOT? If we annotated in the test more, would that help?

There are two problems. I'm fixing one now in this PR, but the other problem requires #102482 (I put it into a separate PR so expect failures here for now).

Comment (MichalStrehovsky) [reply src/libraries/System.Private.Xml/tests/TrimmingTests/System.Private.Xml.TrimmingTests.proj]: @eerhardt do you have a preference on how to proceed here? XmlSerializer doesn't seem to be actively maintained by anyone so nobody is looking at that PR and I would like to get this PR in so we can finally close the tracking issue.

**Comment by MichalStrehovsky** [comment]
@eerhardt this should be ready for review now

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks for the fixes here!

**Pull request merged**